MagazineLayout/Public/UICollectionViewDelegateMagazineLayout.swift

```diff
@@ -135,4 +137,247 @@ public protocol UICollectionViewDelegateMagazineLayout: UICollectionViewDelegate
+    finalLayoutAttributesForRemovedFooterInSectionAtIndex index: Int,
+    byModifying finalLayoutAttributes: UICollectionViewLayoutAttributes)
+
+  ///   Asks the delegate to modify a layout attributes instance so that it represents the final visual state of a background being
+  ///   removed via `UICollectionView.deleteSections(_:)`.
+  ///
+  ///   The `finalLayoutAttributes` instance is a reference type, and therefore can be modified directly. If the provided
+  ///   layout attributes instance is not changed in the implementation of this function, then the background will be removed with no
+  ///   animation.
+  ///
+  ///   - Parameters:
+  ///      - collectionView: The collection view using the layout.
+  ///      - collectionViewLayout: The layout requesting the information.
+  ///      - index: The index of the section containing the background.
+  ///      - finalLayoutAttributes: The unmodified layout attributes representing the final visual state of the removed
+  ///      background. Modify properties on this instance to create a background delete animation.
+  func collectionView(
+    _ collectionView: UICollectionView,
+    layout collectionViewLayout: UICollectionViewLayout,
+    finalLayoutAttributesForRemovedBackgroundInSectionAtIndex index: Int,
+    byModifying finalLayoutAttributes: UICollectionViewLayoutAttributes)
+
+}
+
+// MARK: Default Insert Animations
+
+public extension UICollectionViewDelegateMagazineLayout {
+
+  func collectionView(
+    _ collectionView: UICollectionView,
+    layout collectionViewLayout: UICollectionViewLayout,
+    initialLayoutAttributesForInsertedItemAt indexPath: IndexPath,
+    byModifying initialLayoutAttributes: UICollectionViewLayoutAttributes)
+  {
+    // The default insert animation is a simple fade-in.
+    initialLayoutAttributes.alpha = 0
+  }
+
+  func collectionView(
+    _ collectionView: UICollectionView,
+    layout collectionViewLayout: UICollectionViewLayout,
+    initialLayoutAttributesForInsertedHeaderInSectionAtIndex index: Int,
+    byModifying initialLayoutAttributes: UICollectionViewLayoutAttributes)
+  {
+    // The default insert animation is a simple fade-in.
+    initialLayoutAttributes.alpha = 0
+  }
+
+  func collectionView(
+    _ collectionView: UICollectionView,
+    layout collectionViewLayout: UICollectionViewLayout,
+    initialLayoutAttributesForInsertedFooterInSectionAtIndex index: Int,
+    byModifying initialLayoutAttributes: UICollectionViewLayoutAttributes)
+  {
+    // The default insert animation is a simple fade-in.
+    initialLayoutAttributes.alpha = 0
+  }
+
+  func collectionView(
+    _ collectionView: UICollectionView,
+    layout collectionViewLayout: UICollectionViewLayout,
+    initialLayoutAttributesForInsertedBackgroundInSectionAtIndex index: Int,
+    byModifying initialLayoutAttributes: UICollectionViewLayoutAttributes)
+  {
+    // The default insert animation is a simple fade-in.
+    initialLayoutAttributes.alpha = 0
```

Comment:
You could remove the duplication by passing these attributes into default insert/delete animation functions.